packages/cloudflare/src/cli/build/patches/plugins/next-minimal.ts

```
@@ -0,0 +1,92 @@
+import { patchCode } from "../ast/util.js";
+import { ContentUpdater } from "./content-updater.js";
+
+// We try to be as specific as possible to avoid patching the wrong thing here
+// It seems that there is a bug in the worker runtime. When the AbortController is created outside of the request context it throws an error (not sure if it's expected or not) except in this case.
```

Workers can not do IO outside of the the request - there should be a clear error message in this case (and it should be easy enough to try outside of the context of Next).

Note that we also plan to not run any Next code oustide of a request context which should also help here.

---

Yeah, but in this case this is not actually doing any IO. Just creating an `AbortController` outside of the request context will trigger the error as described here https://github.com/cloudflare/workerd/issues/3657

It's the import of Next itself in this case that is causing trouble. I haven't found why it fails silently in this case

---

Thanks a lot for creating the issue!

FYI The service worker syntax is deprecated

```ts
addEventListener("fetch", event => {
  event.respondWith(new Response("Hello World"));
});
```

See https://developers.cloudflare.com/workers/reference/migrate-to-module-workers